openedx-unsupported/ecommerce
# REV-2687: update get_receipt_page_url to return ecommerce or MFE versions of the page

[REV-2687](https://2u-internal.atlassian.net/browse/REV-2687)

We want get_receipt_page_url to now be able to return the url for the original/live receipt page (in ecommerce) OR the new receipt page being rebuilt in frontend-app-ecommerce in REV-2683. To determine which one to return, the function will look at (a) a new `enable_receipts_via_ecommerce_mfe` waffle flag and (b) a new `ECOMMERCE_MICROFRONTEND_URL` variable. 

With this PR, the code will: 
- look for an environment variable for `ECOMMERCE_MICROFRONTEND_URL` and import it in settings, which will then be imported into the other files that use it
- check the value of the new `enable_receipts_via_ecommerce_mfe` waffle flag, and send that along when calling get_receipt_page_url. (Exception: I didn't update the send_mail function in the EnrollmentCodeFulfillmentModule: it's an email that goes out when someone has purchased enrollment codes. In this case- or any where the waffle flag value is not passed- we default to False aka show the original ecommerce receipt page). 
- then in get_receipt_page_url, we use both the waffle flag value and the presence of that `ECOMMERCE_MICROFRONTEND_URL` environment variable to determine which url to return (if both are true/present, give the new one, otherwise give the old one)

## Testing instructions for this branch
- To start, place an order and see that afterward we're redirected to the old receipt page
- (then unenroll to be able to purchase that demo course again) 
- In your [local ecommerce admin page](http://localhost:18130/admin/waffle/flag/5/change/), add waffle flag set to enabled for everyone: `enable_receipts_via_ecommerce_mfe`
- place an order again, and see that you're redirected to the url for the new page (coming soon in REV-2683, http://localhost:1996/receipt/?order_number=EDX-100020&disable_back_button=1, but tested earlier with the order history page since it's available)
- you can also exercise this code by running the following test in ecommerce-shell: pytest ecommerce/extensions/checkout/tests/test_utils.py. Initially there is no environment variable set, so it will go to the original receipt page in all three cases, but then if you set a local env variable then get_receipt_page_url will return the new url. 


## Required Testing
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)
